### PR TITLE
[ESIMD] Fix compilation errors caused by type_name() function

### PR DESCRIPTION
**Pull request by v-klochkov**
The usage of static for one of utility functions caused such warnings before: function 'esimd_test::type_name<unsigned long>' has internal linkage
      but is not defined [-Wundefined-internal]

Fixes https://github.com/intel/llvm/issues/8934

### CODE REVIEW

**Review by turinevgeny — APPROVED** [review]

**Comment by v-klochkov** [comment]
Please wait, don't merge. The fix is potentially not complete. 
The type printing function also causes errors during linking phase even if remove 'static' keyword. Investigating.

File: sycl/test-e2e/ESIMD/esimd_test_utils.hpp [hunk 1]
```diff
@@ -568,7 +568,7 @@ std::unique_ptr<T, USMDeleter> usm_malloc_shared(queue q, int n) {
   return std::move(res);
 }
 
-template <class T> static const char *type_name();
+template <class T> const char *type_name() { return typeid(T).name(); }
```
Comment (turinevgeny) [inline sycl/test-e2e/ESIMD/esimd_test_utils.hpp:571]: If the function is defined, then likely the warning will not be issued.

Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/esimd_test_utils.hpp:571]: yes, but that 'static' keyword did not seem to be needed for this template function

**Comment by v-klochkov** [comment]
The fail on AMDGPU is unrelated:
SYCL :: AtomicRef/compare_exchange_generic_local.cpp

The fail in Jenkins/Precommit (ProtexIP) showed pass after restart: https://cje-fm-owrp-prod03.devtools.intel.com/satg-ce-compiler-pipelines/blue/organizations/jenkins/SYCL_CI%2Fintel%2FProtexIP/detail/ProtexIP/188/pipeline/

**Pull request merged**